### PR TITLE
Cleanup v1alpha1 Subscription codebase - Part 1

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Cleanup v1alpha1 Subscription codebase - Part 1

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/16310

### CODE REVIEW

File: components/eventing-controller/pkg/backend/jetstreamv2/config_internal_unit_test.go [hunk 1]
```diff
@@ -65,131 +62,6 @@ func TestUnitValidate_For_Errors(t *testing.T) {
 		})
 	}
 }
-func Test_GetNATSConfig(t *testing.T) {
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/jetstreamv2/config_internal_unit_test.go:68]: Moved to `pkg/env`

File: components/eventing-controller/pkg/backend/jetstreamv2/jetstream_integration_test.go [hunk 2]
```diff
@@ -475,9 +473,220 @@ func TestJetStream_NATSSubscriptionCount(t *testing.T) {
 	}
 }
 
-func defaultNATSConfig(url string, port int) backendnats.Config {
+// TestJetStream_ServerRestart tests that eventing works when NATS server is restarted
+// for scenarios involving the stream storage type and when reconnect attempts are exhausted or not.
+func TestJetStream_ServerRestart(t *testing.T) { //nolint:gocognit
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/jetstreamv2/jetstream_integration_test.go:478]: Copied from [here](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/nats/jetstream/jetstream_v1v2_integration_test.go)

File: components/eventing-controller/pkg/backend/jetstreamv2/jetstream_integration_test.go [hunk 3]
```diff
@@ -475,9 +473,220 @@ func TestJetStream_NATSSubscriptionCount(t *testing.T) {
...
+			require.NoError(t, SendEventToJetStream(testEnvironment.jsBackend, ev2data))
+			expectedEv2Data := fmt.Sprintf("%q", ev2data)
+			require.NoError(t, subscriber.CheckEvent(expectedEv2Data))
+		})
+	}
+}
+
+// TestJetStream_ServerAndSinkRestart tests that the messages persisted (not ack'd) in the stream
+// when the sink is down reach the subscriber even when the NATS server is restarted.
+func TestJetStream_ServerAndSinkRestart(t *testing.T) {
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/jetstreamv2/jetstream_integration_test.go:594]: Copied from [here](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/nats/jetstream/jetstream_v1v2_integration_test.go)

File: components/eventing-controller/pkg/backend/jetstreamv2/test_helpers.go [hunk 4]
```diff
@@ -36,17 +37,32 @@ type TestEnvironment struct {
 	logger     *logger.Logger
 	natsServer *server.Server
 	jsClient   *jetStreamClient
-	natsConfig backendnats.Config
+	natsConfig env.NATSConfig
 	cleaner    cleaner.Cleaner
 	natsPort   int
 }
 
+func StartNATSServer(serverOpts ...evtesting.NatsServerOpt) (*server.Server, int, error) {
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/jetstreamv2/test_helpers.go:45]: Copied from [here](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/nats/testing/test_helpers.go)

File: components/eventing-controller/pkg/backend/jetstreamv2/test_helpers.go [hunk 5]
```diff
@@ -36,17 +37,32 @@ type TestEnvironment struct {
...
+	natsPort, err := evtesting.GetFreePort()
+	if err != nil {
+		return nil, 0, err
+	}
+	serverOpts = append(serverOpts, evtesting.WithPort(natsPort))
+	natsServer := evtesting.RunNatsServerOnPort(serverOpts...)
+	return natsServer, natsPort, nil
+}
+
+func NewNatsMessagePayload(data, id, source, eventTime, eventType string) string {
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/jetstreamv2/test_helpers.go:55]: Copied from [here](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/nats/testing/test_helpers.go)

File: components/eventing-controller/pkg/backend/sink/validator.go [hunk 6]
```diff
@@ -2,88 +2,53 @@ package sink
 
 import (
 	"context"
```
Comment (mfaizanse) [inline components/eventing-controller/pkg/backend/sink/validator.go:4]: Replaced this file by: https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/pkg/backend/sink/v2/validator.go

File: components/eventing-controller/Makefile
Comment (k15r) [inline components/eventing-controller/Makefile:1]: please also update lines 123 ff. There is still a check for the new crd version

Comment (mfaizanse) [reply components/eventing-controller/Makefile:1]: updated

**Review by k15r — APPROVED** [review]

**Pull request merged**